src/store/modules/geometry/helpers.js

```diff
@@ -283,31 +283,52 @@ const helpers = {
     /*
      * return the set of saved vertices directly on an edge, not including edge endpoints
      */
-  splittingVerticesForEdgeId(edge_id, geometry, spacing) {
-    const edge = geometry.edges.find(e => e.id === edge_id),
-      edgeV1 = this.vertexForId(edge.v1, geometry),
-      edgeV2 = this.vertexForId(edge.v2, geometry);
-      // look up all vertices touching the edge, ignoring the edge's endpoints
-    const verticesToSplit = geometry.vertices.filter((vertex) => {
+  splittingVerticesForEdgeId(edge, geometry, spacing, vertices) {
+    // Use a memoized cache of all the vertices to quickly find the necessary vertices
+    // If the vertex can't be found in the map, fall back to searching the entire array with find
```

Comment:
What's the situation when a vertex isn't found in `vertexMap`, but _is_ found in `geometry.vertices`?

Reply:
Whenever it has to split an edge and generate new vertices it will miss; it will also miss when a new space overlaps with an old space and new face points are generated.